### ISSUE TITLE
Front end-issue with umbracoforms.js

### ISSUE THREAD

**Issue by enkelmedia** [umbraco/Umbraco.Forms.Issues#319]
I'm having problems with front end exceptions in umbracoforms.js,

![image](https://user-images.githubusercontent.com/1782524/78244877-c125c700-74e6-11ea-914a-7057d8953044.png)

Looks like this has been raised before in https://github.com/umbraco/Umbraco.Forms.Issues/issues/273 and https://github.com/umbraco/Umbraco.Forms.Issues/issues/295 but they are marked as fixed for 8.4.0 which is not the case.

This line still looks like this in 8.4.0
```javascript
init({ form: JSON.parse(decodeURI(formsCollection[i])) });
```

Since formsCollection[i] is already a valid JSON-object this throws an error. I guess this is some backwards-compatability-thing? In that case you might want to check if the variable is string or already a valid object before calling init.

Ie:
```javascript
if (typeof formsCollection[i] === 'object') {
    init({ form: formsCollection[i] });
} else {
    init({ form: JSON.parse(decodeURI(formsCollection[i])) });
}
```

/ m

**Comment by nul800sebastiaan** [comment]
@enkelmedia Did you also update `Views/Partials/Forms/Themes/default/Script.cshtml`? That's where I believe the fix is, Line 20 should read: `umbracoFormsCollection.push('@Html.Raw(Uri.EscapeUriString(JsonConvert.SerializeObject(formJsObj)))');`

The line you're pointing to was changed from `init({ form: JSON.parse(unescape(formsCollection[i])) });` to `init({ form: JSON.parse(decodeURI(formsCollection[i])) });`.

I am sure you're correct, I didn't try to read this code, so just to confirm your Script.cshtml is also up to date?

**Comment by AndyButland** [comment]
Looks like this has also been reported at #239 and #522, and should be fixed with the 8.7.0 release.
